vizro-core/tests/component_library/test_component_library.py

```diff
@@ -0,0 +1,93 @@
+# ruff: noqa: F403, F405
+import dash_bootstrap_components as dbc
+import pandas as pd
+import pytest
+from dash import Dash, html
+from tests.helpers.common import compare_images
+
+from vizro.figures.library import kpi_card, kpi_card_reference
+
+df_kpi = pd.DataFrame(
+    {
+        "Actual": [100, 200, 700],
+        "Reference": [100, 300, 500],
+        "Category": ["A", "B", "C"],
+    }
+)
+
+example_cards = [
+    kpi_card(data_frame=df_kpi, value_column="Actual", title="KPI with value"),
+    kpi_card(
+        data_frame=df_kpi,
+        value_column="Actual",
+        title="KPI with aggregation",
+        agg_func="median",
+    ),
+    kpi_card(
+        data_frame=df_kpi,
+        value_column="Actual",
+        title="KPI formatted",
+        value_format="${value:.2f}",
+    ),
+    kpi_card(
+        data_frame=df_kpi,
+        value_column="Actual",
+        title="KPI with icon",
+        icon="shopping_cart",
+    ),
+]
+
+example_reference_cards = [
+    kpi_card_reference(
+        data_frame=df_kpi,
+        value_column="Actual",
+        reference_column="Reference",
+        title="KPI ref. (pos)",
+    ),
+    kpi_card_reference(
+        data_frame=df_kpi,
+        value_column="Actual",
+        reference_column="Reference",
+        agg_func="median",
+        title="KPI ref. (neg)",
+    ),
+    kpi_card_reference(
+        data_frame=df_kpi,
+        value_column="Actual",
+        reference_column="Reference",
+        title="KPI ref. formatted",
+        value_format="{value}€",
+        reference_format="{delta}€ vs. last year ({reference}€)",
+    ),
+    kpi_card_reference(
+        data_frame=df_kpi,
+        value_column="Actual",
+        reference_column="Reference",
+        title="KPI ref. with icon",
+        icon="shopping_cart",
+    ),
+]
+
+
+@pytest.mark.filterwarnings("ignore:HTTPResponse.getheader():DeprecationWarning")
+@pytest.mark.filterwarnings("ignore::pytest.PytestUnhandledThreadExceptionWarning")
+@pytest.mark.filterwarnings("ignore:unclosed file:ResourceWarning")
+def test_kpi_card(dash_duo):
+    app = Dash(__name__, external_stylesheets=[dbc.themes.BOOTSTRAP])
+    app.layout = dbc.Container(
+        [
+            html.H1(children="KPI Cards"),
+            dbc.Stack(
+                children=[
+                    dbc.Row([dbc.Col(kpi_card) for kpi_card in example_cards]),
+                    dbc.Row([dbc.Col(kpi_card) for kpi_card in example_reference_cards]),
+                ],
+                gap=4,
+            ),
+        ]
+    )
+    dash_duo.start_server(app)
+    dash_duo.wait_for_page(timeout=20)
+    dash_duo.wait_for_element("div[class='card-kpi card']")
+    compare_images(dash_duo.driver, "base_kpi_comp_lib.png", "tests_kpi_comp_lib")
```

Comment:
Not important for now, but maybe something to tweak later: why do we need to pass in file names manually here? Can't you just label them automatically based on the called `node`? I saw you do something similar in the `dash_br` PR.